### ISSUE TITLE
Add more complex examples

### ISSUE THREAD

**Issue by bog-dan-ro** [conan-io/examples2#41]
The existing examples are great to get started with simple projects, but are not enough for more complex projects:
- cmake/autotools/meson projects with multiple targets (e.g mpg123, vorbis, etc.). 
- According to https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-scm-attribute the scm attribute was removed but [Git ](https://docs.conan.io/en/latest/reference/conanfile/tools/scm/git.html#conan-tools-scm-git) seems to be missing a few features like `"submodule": "shallow",` `"shallow": True,` scm attributes. Again, a more complex example will be more than welcome.

**Comment by jngrb** [comment]
I agree. Further best-practices example that I would like to see are these:

- Using `export` and `source` (instead of the old `scm` attribute) but where the conanfile (and the CMakeLists.txt) are not in the root of the Git repo.
  I.e. the existing multi-subprojects example [conanfile/layout/multiple_subprojects](https://github.com/conan-io/examples2/tree/main/examples/conanfile/layout/multiple_subprojects) but just with sources from Git.

- Using `export` and `source` (instead of the old `scm` attribute) but where the source files are distributed across several Git submodules.
